lntest/itest/macaroons.go

```diff
@@ -166,3 +168,178 @@ func testMacaroonAuthentication(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("returned address was not a regtest address")
 	}
 }
+
+// testBakeMacaroon checks that when creating macaroons, the permissions param
+// in the request must be set correctly, and the baked macaroon has the intended
+// permissions.
+func testBakeMacaroon(net *lntest.NetworkHarness, t *harnessTest) {
```

Comment:
Nice test, thanks for adding that! Some of these test cases should have been added when the original bake macaroon functionality was added. So it's nice to have them now.